src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -79,4 +80,14 @@ public void storeMessage(Message message) {
 
     return messages;
   }
+
+  /**
+   * Gets total number of messages posted.
+   * @return an integer respresenting the total numner of messages posted on the webpage.
+   */
+  public int getTotalMessageCount() {
+    Query query = new Query("Message");
+    PreparedQuery results = datastore.prepare(query);
```

Comment:
You define query on the previous line and pass it in here, but make no other changes to the object.

You could pass in the new Query("Message") directly into datastore.prepare() but you could also keep it as is, what's your reasoning for the object definition?

Reply:
You're right! I was just following the tutorials